integration_test.go

```diff
@@ -873,6 +874,40 @@ func TestRemovalOfWatch(t *testing.T) {
 	wg.Wait()
 }
 
+func TestRemovalOfWatchTwice(t *testing.T) {
+	// Create directory to watch
+	testDir := tempMkdir(t)
+	defer os.RemoveAll(testDir)
+
+	// Create a file before watching directory
+	testFileAlreadyExists := filepath.Join(testDir, "TestFsnotifyEventsExisting.testfile")
+	{
+		var f *os.File
+		f, err := os.OpenFile(testFileAlreadyExists, os.O_WRONLY|os.O_CREATE, 0666)
+		if err != nil {
+			t.Fatalf("creating test file failed: %s", err)
+		}
+		f.Sync()
+		f.Close()
+	}
+
+	watcher := newWatcher(t)
+	defer watcher.Close()
+
+	addWatch(t, watcher, testDir)
+
+	var err error
+	if err = watcher.Remove(testDir); err != nil {
+		t.Fatalf("Could not remove the watch: %v\n", err)
+	}
+	if err = watcher.Remove(testDir); err == nil {
+		t.Fatalf("Shold: %v\n", err)
```

Comment:
Perhaps `t.Fatalf("Removal of already removed watcher should have failed")`

Reply:
fixed